### DISCUSSION TITLE
What is the development & ownership model for this repo?

### DISCUSSION THREAD

**Discussion by ahelwer** [nvim-treesitter/nvim-treesitter#1663] [Q&A] [answered]
Say I have a tree-sitter grammar for a language I'd like to add to nvim-treesitter. I would add query files in `queries/languagename` and a filetype detection script in `ftdetect/languagename.vim`. Would I have to get signoff on a PR every time I want to update those query files? What about if I update the grammar itself? Are the grammars referenced as submodules or something?

**Comment by stsewd** [comment, marked as answer]
Yes, you'll need to update the queries here and wait for review, and you can be the maintainer of that language, so we would pin you if there is something that needs your expertise.

> What about if I update the grammar itself? Are the grammars referenced as submodules or something?

The grammars are pinned to a specific commit https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lockfile.json, and we have a bot that updates them. If our CI breaks because of an update, we usually contact the maintainer of that language to update the query files or fix it by ourselves if it's something easy.
